odiglet/pkg/kube/runtime_details/instrumentationconfigs_controller.go

```diff
@@ -0,0 +1,136 @@
+package runtime_details
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	odigosv1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	k8sutils "github.com/odigos-io/odigos/k8sutils/pkg/utils"
+	kubeutils "github.com/odigos-io/odigos/odiglet/pkg/kube/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type instrumentationConfigPredicate struct{}
+
+func (p *instrumentationConfigPredicate) Create(e event.CreateEvent) bool {
+	obj, ok := e.Object.(*odigosv1.InstrumentationConfig)
+	if !ok {
+		return false
+	}
+	// we only care about new InstrumentationConfig objects
+	// the event will be triggered also when odiglet starts, and the instrumentation config is created
+	// in controller-runtime cache
+	// checking the RuntimeDetailsByContainer map should filter cases where we do not want to trigger the reconcile
+	// for workload which has already been reconciled previously
+	return len(obj.Status.RuntimeDetailsByContainer) == 0
+}
+
+func (p *instrumentationConfigPredicate) Update(e event.UpdateEvent) bool {
+	return false
+}
+
+func (p *instrumentationConfigPredicate) Delete(e event.DeleteEvent) bool {
+	return false
+}
+
+func (p *instrumentationConfigPredicate) Generic(e event.GenericEvent) bool {
+	return false
+}
+
+type InstrumentationConfigReconciler struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	Clientset *kubernetes.Clientset
```

Comment:
What are the main reasons for using this client? saving cache space? correctness of the observed generations? If it is the latter, does it really affect correctness or does the cache have guarantees that are enough for us?
In any case. worth adding a comment about why this client is used.